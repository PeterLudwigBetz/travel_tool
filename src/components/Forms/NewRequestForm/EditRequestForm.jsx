import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
import Script from 'react-load-script';
import { FormContext } from '../FormsAPI';
import PersonalDetailsFieldset from './FormFieldsets/PersonalDetails';
// import TravelDetailsFieldset from './FormFieldsets/TravelDetails';
import EditInputFieldSet from './FormFieldsets/EditInputFieldSet';
import SubmitArea from './FormFieldsets/SubmitArea';
import './NewRequestForm.scss';

class EditRequestForm extends PureComponent {
  constructor(props) {
    super(props);
    const { request } = this.props; 
    console.log(request);

    this.defaultState = {
      values: {
        name: request.name, // FIX: need to be refactor later
        gender: request.gender,
        department: request.department,
        role: request.role,
        manager: request.manager,
      },
      trips: request.trips,
      errors: {},
      hasBlankFields: true,
      checkBox: 'notClicked',
      selection: 'return',
      collapse: false,
      title: 'Hide Details',
      position: 'none',
      line: '1px solid #E4E4E4',
      parentIds: 1
    };
    this.state = { ...this.defaultState };
  }

  componentDidMount() {
    // console.log(this.props);
  }
 
  componentWillUnmount() {
    this.handleClearForm();
  }
  // an onChange handler will be created by the Input component when it's rendered
  

  handleSubmit = event => {
    event.preventDefault();
    const { handleEditRequest } = this.props;
    const { values } = this.state;
    const checkBoxState = localStorage.getItem('state');
    if (checkBoxState === 'clicked'){
      const [name, gender, department, role, manager] = [values.name, values.gender, values.department, values.role, values.manager];
      this.savePersonalDetails(name,gender,department,role,manager);
    }
    if (this.validate()) {
      // call create the request
      let data = { ...values };
      if (data.destination === 'Other') {
        data.destination = data.otherDestination;
      }
      delete data.otherDestination;
      handleEditRequest('vLT9a_VEr', data);
    }
  };
   handleClearForm = () => {
     this.setState({ ...this.defaultState });
   };
   validate = field => {
     let { values, errors } = this.state;
     [errors, values] = [{ ...errors }, { ...values }];
     let hasBlankFields = false;
     // check if to enforce otherDestination
     const requireOtherDestination = values.destination === 'Other';
     if (!requireOtherDestination) delete values['otherDestination'];
     !values[field]
       ? (errors[field] = 'This field is required')
       : (errors[field] = '');
     // check if the form has any other blank fields
     // this will qualify the form as fully filled or not
     hasBlankFields = Object.keys(values).some(key => !values[key]);
     // update the form's validity and return a boolean to use on Submit
     this.setState(prevState => {
       return { ...prevState, errors, hasBlankFields };
     });
     return !hasBlankFields;
   };
   
   render() {
     const { values, errors, hasBlankFields, selection, parentIds, trips } = this.state;
     const { managers, creatingRequest, request, } = this.props;
     const { id } = request;
     //  console.log(id);
     return (
       <FormContext targetForm={this} errors={errors} validatorName="validate">
         {creatingRequest && (
           <h5 style={{display: 'flex', justifyContent: 'center', fontFamily: 'DIN Pro'}}>
          Creating request...
           </h5>
         )}
         <form onSubmit={this.handleSubmit} className="new-request">
           <PersonalDetailsFieldset values={values} managers={managers} />
           {trips.map(id => (
             <EditInputFieldSet
               key={trips.id} 
               values={values}
               value="232px" 
               selection={selection}
               handleDate={this.onChangeDate}
               handleChange={this.handleRadioButton}
               onChangeInput={this.onChangeInput}
               parentIds={parentIds}
               addNewTrip={this.addNewTrip}
               removeTrip={this.removeTrip} 
             />
           ))}
           <hr />
           <Script url={process.env.REACT_APP_CITY} />   
           <SubmitArea
             onCancel={this.handleClearForm}
             hasBlankFields={hasBlankFields}
             send="Update Request"
           />
         </form>
       </FormContext>
     );
   }
}
EditRequestForm.propTypes = {
  handleEditRequest: PropTypes.func.isRequired,
  managers: PropTypes.array,
  creatingRequest: PropTypes.bool,
};
EditRequestForm.defaultProps = {
  creatingRequest: false,
  managers: []
};
export default EditRequestForm;

