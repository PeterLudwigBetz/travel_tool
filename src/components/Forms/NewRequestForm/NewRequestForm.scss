form.new-request {
  fieldset {
    .input-group {
      display: flex;
    }
    .spaces {
      margin-right: 30px;
    }

    .line {
      width: 100%;
      padding-bottom: 11px;
    }
    .style-h5 {
      display: "flex";
      justify-content: "center";
      font-family: "DIN Pro";
    }

    .hide-details {
      height: 18px;
      width: 103px;
      color: #4f4f4f;
      font-family: "DIN Pro";
      font-size: 14px;
      font-weight: 500;
      line-height: 18px;
      float: right;
      cursor: pointer;
    }
    .expand {
      float: right;
      margin: 7px 0px 0px 6px;
    }

    .rectangle {
      height: 95px;
      width: 100%;
      border: 1px solid #e4e4e4;
      border-radius: 2px;
      background-color: #f8f8f8;
      margin: 7px 0px 20px 0px;
      color: gray;
    }
    .style-details {
      display: flex;
      padding: 10px;
      margin-left: 20px;
    }

    .travel-to {
      margin-right: 20px;
      position: relative;
      width: 198px;
    }

    .location-icon {
      position: absolute;
      right: 15px;
      top: 39px;
    }

    .others-width {
      margin-right: 20px;
    }
  }
}
