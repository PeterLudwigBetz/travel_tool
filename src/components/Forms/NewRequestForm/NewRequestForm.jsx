import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
import moment from 'moment';
import { FormContext } from '../FormsAPI';
import PersonalDetailsFiedset from './FormFieldsets/PersonalDetails';
import TravelDetailsFiedset from './FormFieldsets/TravelDetails';
import SubmitArea from './FormFieldsets/SubmitArea';
import './NewRequestForm.scss';

class NewRequestForm extends PureComponent {
  constructor(props) {
    super(props);
    const user = props.user;

    this.defaultState = {
      values: {
        name: user ? user.UserInfo.name : '', // FIX: need to be refactor later
        gender: '',
        department: '',
        role: '',
        manager: '',
        origin: '',
        destination: '',
        otherDestination: '',
        departureDate: null,
        arrivalDate: null
      },
      errors: {},
      hasBlankFields: true
    };

    this.state = { ...this.defaultState };
  }


  componentWillUnmount() {
    this.handleClearForm();
  }
  // an onChange handler will be created by the Input component when it's rendered

  handleSubmit = event => {
    event.preventDefault();
    const { handleCreateRequest } = this.props;
    const { values } = this.state;
    
    if (this.validate()) {
      // call create the request
      let data = values;
      
      if (data.destination === 'Other') {
        data.destination = data.otherDestination;
      }
      delete data.otherDestination;
      handleCreateRequest(data);
    }
  };

  handleClearForm = () => {
    this.setState({ ...this.defaultState });
  };

  validate = field => {
    let { values, errors } = this.state;
    [errors, values] = [{ ...errors }, { ...values }];
    let hasBlankFields = false;

    // check if to enforce otherDestination
    const requireOtherDestination = values.destination === 'Other';
    if (!requireOtherDestination) delete values['otherDestination'];

    !values[field]
      ? (errors[field] = 'This field is required')
      : (errors[field] = '');

    // check if the form has any other blank fields
    // this will qualify the form as fully filled or not
    hasBlankFields = Object.keys(values).some(key => !values[key]);
    // update the form's validity and return a boolean to use on Submit
    this.setState(prevState => {
      return { ...prevState, errors, hasBlankFields };
    });
    return !hasBlankFields;
  };

  render() {
    const { values, errors, hasBlankFields } = this.state;
    const { user, creatingRequest } = this.props;

    return (
      <FormContext targetForm={this} errors={errors} validatorName="validate">

        {creatingRequest && (
          <h5 style={{display: 'flex', justifyContent: 'center', fontFamily: 'DIN Pro'}}>
          Creating request...
          </h5>
        )}
        <form onSubmit={this.handleSubmit} className="new-request">
          <PersonalDetailsFiedset values={values} />
          <TravelDetailsFiedset values={values} />
          <hr />
          <SubmitArea
            onCancel={this.handleClearForm}
            hasBlankFields={hasBlankFields}
          />
        </form>
      </FormContext>
    );
  }
}

NewRequestForm.propTypes = {
  handleCreateRequest: PropTypes.func.isRequired,
  user: PropTypes.object,
  creatingRequest: PropTypes.bool,
};

NewRequestForm.defaultProps = {
  creatingRequest: false,
  user: {}
};

export default NewRequestForm;
