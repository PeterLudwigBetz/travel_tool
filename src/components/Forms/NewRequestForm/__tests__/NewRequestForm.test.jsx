import React from 'react';
import sinon from 'sinon';
import NewRequestForm from '../NewRequestForm';
import PersonalDetailsFieldset from '../FormFieldsets/PersonalDetails';

describe('<NewRequestForm />', () => {
  let wrapper, onSubmit;
  onSubmit = jest.fn();

  window.google ={
    maps:{
      Marker:class{},
      Map:class{ setTilt(){} fitBounds(){}},
      LatLngBounds:class{},
      places: {
        Autocomplete: class {
          addListener () {
            return {
              calledOnce: function () {
                jest.fn();
              },
            };
          }
          calledOnce () {
            return {
              calledOnce: function () {
                jest.fn();
              },
            };
          }
        }
      },
      AutocompleteService:class{},
    },
    MarkerClusterer:class{},
    Geocoder:class{},
  };


  const props = {
    url: 'http://www.home.com',
    process: 'process.emv.REACT_APP_ANDELA_AUTH_HOST',
    user: {
      UserInfo: {
        name: 'John Doe'
      },
    },
    google: {},
    loading: false,
    errors: [],
    handleCreateRequest: jest.fn(() => {}),
    updateUserProfile: jest.fn(() => {}),
    managers: [
      {
        fullName: 'Test User',
        email: 'test.user@andela.com'
      }
    ]
  };
  const event = {
    preventDefault: jest.fn(),
    google: jest.fn(),
    target: {
      dataset: {
        parentid: 'destination-0'
      },
      name: 'destination-0'
    },
    nativeEvent: {
      path: [0,1,2,3,4,5,6,{id: 'departure-0_date'}]
    }
  };
  window.date = {
    format: jest.fn(),
  }

  const {user} = props;
  const defaultState = {
    values: {
      name: '', // FIX: need to be refactor later
      gender: '',
      department: '',
      role: '',
      manager: '',
    },
    trips: [{}],
    errors: {},
    collapse: false,
    hasBlankFields: true,
    checkBox: 'notClicked',
    selection: 'return'
  };


  beforeEach(() => {
    wrapper = mount(<NewRequestForm {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders three fieldsets', () => {
    const fieldsets = wrapper.find('fieldset');
    expect(fieldsets).toHaveLength(3);
  });

  it('renders the personal details fieldset', () => {
    const personalDetails = wrapper.find('fieldset.personal-details');
    expect(personalDetails).toHaveLength(1);
  });

  it('renders the travel details fieldset', () => {
    const personalDetails = wrapper.find('fieldset.travel-details');
    expect(personalDetails).toHaveLength(1);
  });

  it('renders four buttons', () => {
    const buttons = wrapper.find('button');
    expect(buttons).toHaveLength(4);
  });

  it('picks input values', () => {
    wrapper.find('input[name="name"]').simulate('change', {
      target: {
        name: 'name',
        value: 'John Mutuma'
      }
    });
    expect(wrapper.state().values.name).toBe('John Mutuma');
  });

  it('validates input on blur', () => {
    wrapper.find('input[name="name"]').simulate('blur');
    wrapper.update();
    expect(wrapper.state().errors.name).toBe('This field is required');
  });

  it('validates form before sending data', () => {
    const form = wrapper.find('form');
    form.simulate('submit');
    expect(onSubmit).toHaveBeenCalledTimes(0);
  });


  it('set destination on onchange', () => {
    let destination = wrapper.find('input[name="destination-0"]');
    destination.simulate('change', {
      target: {
        name: 'lagos',
        dataset: {
          value: 'destination-0'
        },
        value: 'Lagos'
      }
    });
    expect(wrapper.state().values.name).toBe('');
  });

  it('set date on onchange', () => {
    let destination = wrapper.find('input[name="origin-0"]');
    destination.simulate('change', {
      target: {
        name: 'lagos',
        dataset: {
          value: 'origin-0'
        },
        value: 'Lagos'
      }
    });
    expect(wrapper.state().values.name).toBe('');
  });

  it('set date on onchange', () => {
    let destination = wrapper.find('input[name="departureDate-0"]');
    destination.simulate('change', {
      target: {
        name: 'lagos',
        dataset: {
          value: 'departureDate-0'
        },
        value: 'Lagos'
      }
    });
    expect(wrapper.state().values.name).toBe('');
  });

  it('calls onChangeInput when input is changed in state', () => {
    const shallowWrapper = shallow(<NewRequestForm {...props} />);
    sinon.spy(shallowWrapper.instance(), 'onChangeInput');
    shallowWrapper.instance().onChangeInput(event);
    expect(shallowWrapper.instance().onChangeInput.calledOnce).toEqual(true);
  });

  it('call event and resets return date when departure date is changed', () => {
    const shallowWrapper = shallow(<NewRequestForm {...props} />);
    sinon.spy(shallowWrapper.instance(), 'onChangeDate');
    const spy = jest.spyOn(shallowWrapper.instance(), 'resetTripArrivalDate')
    const date = {
      format: () => '2018-11-11'
    }
    shallowWrapper.instance().onChangeDate(date, event);
    expect(shallowWrapper.instance().onChangeDate.calledOnce).toEqual(true);
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('call event when date is changed', () => {
    const shallowWrapper = shallow(<NewRequestForm {...props} />);
    const event = {
      preventDefault: jest.fn(),
      target: {
        dataset: {
          parentid: 'destination-0'
        }
      },
      nativeEvent: {
        path: [0,1,2,3,4,5,6,{id: 'arrival-0_date'}]
      }
    };

    sinon.spy(shallowWrapper.instance(), 'onChangeDate');
    const date = {
      format: () => '2018-12-01'
    }
    shallowWrapper.instance().onChangeDate(date, event);
    expect(shallowWrapper.instance().onChangeDate.calledOnce).toEqual(true);
  });


  it('should change the radio buttton on click to multi ', () => {
    const shallowWrapper = shallow(<NewRequestForm {...props} />);

    const event = {
      preventDefault: jest.fn(),
      target: {
        value: 'multi'
      },
    };
    sinon.spy(shallowWrapper.instance(), 'handleRadioButton');
    shallowWrapper.instance().handleRadioButton(event);
    expect(shallowWrapper.instance().handleRadioButton.calledOnce).toEqual(true);
  });

  it('should change the radio button on click to return ', () => {
    const shallowWrapper = shallow(<NewRequestForm {...props} />);
    const event = {
      preventDefault: jest.fn(),
      target: {
        value: 'return'
      },
    };
    sinon.spy(shallowWrapper.instance(), 'handleRadioButton');
    shallowWrapper.instance().handleRadioButton(event);
    expect(shallowWrapper.instance().handleRadioButton.calledOnce).toEqual(true);
  });

  it('should change the radio button on click to return ', () => {
    const shallowWrapper = shallow(<NewRequestForm {...props} />);
    shallowWrapper.setState({
      selection: 'return'
    })
    const event = {
      preventDefault: jest.fn(),
      target: {
        value: 'multi'
      },
    };
    sinon.spy(shallowWrapper.instance(), 'handleRadioButton');
    shallowWrapper.instance().handleRadioButton(event);
    expect(shallowWrapper.instance().handleRadioButton.calledOnce).toEqual(true);
  });

  it('should update state when a trip is added and when it\'s removed', () => {
    expect.assertions(7);
    wrapper.instance().addNewTrip();
    wrapper.instance().addNewTrip();
    expect(wrapper.instance().state.values[`origin-1`]).toBe('');
    expect(wrapper.instance().state.values[`origin-2`]).toBe('');
    expect(wrapper.instance().state.parentIds).toBe(3);
    wrapper.instance().removeTrip(1);
    expect(wrapper.instance().state.parentIds).toBe(2);
    // still have one trip, origin-1 should now be what was at origin-2 and so forth
    expect(wrapper.instance().state.values[`origin-1`]).toBe('');
    // after shifting state values, origin-{parentIds} in state should be undefined
    expect(wrapper.instance().state.values[`origin-2`]).toBe(undefined);
    expect(wrapper.instance().state.trips).toHaveLength(2);
  });

  it('should close the personal field on button click ', () => {
    const shallowWrapper = shallow(<NewRequestForm {...props} />);
    shallowWrapper.setState({
      collapse: true,
      title: 'Show Details',
      position: 'rotate(266deg)',
      line: 'none'
    })
    sinon.spy(shallowWrapper.instance(), 'collapsible');
    shallowWrapper.instance().collapsible(event);
    expect(shallowWrapper.instance().collapsible.calledOnce).toEqual(true);
  });


  it('selects female gender on button click', () => {
    let femaleButton = wrapper.find('button[data-value="Female"]');
    femaleButton.simulate('click', {
      target: {
        name: 'gender',
        getAttribute: () => {},
        dataset: {
          value: 'Female'
        }
      }
    });

    expect(wrapper.state('values').gender).toBe('Female');
  });

  it('selects male gender on button click', () => {
    let maleButton = wrapper.find('button[data-value="Male"]');
    maleButton.simulate('click', {
      target: {
        name: 'gender',
        getAttribute: () => {},
        dataset: {
          value: 'Male'
        }
      }
    });

    expect(wrapper.state('values').gender).toBe('Male');
  });

  it('should change the radio button on click', () => {
    let maleButton = wrapper.find('button[data-value="Male"]');
    maleButton.simulate('click', {
      target: {
        name: 'gender',
        getAttribute: () => {},
        dataset: {
          value: 'Male'
        }
      }
    });
    expect(wrapper.state('values').gender).toBe('Male');
  });



  it('selects gender with browsers that do not support custom datasets', () => {
    let femaleButton = wrapper.find('button[data-value="Female"]');
    femaleButton.simulate('click', {
      target: {
        name: 'gender',
        getAttribute: attrib => 'Female'
      }
    });

    expect(wrapper.state('values').gender).toBe('Female');
  });

  it('should not toggle the modal if request submission failed', () => {
    wrapper.setState({
      values: {
        name: 'test',
        gender: 'test',
        department: 'test',
        role: 'test',
        manager: 'test',
        origin: 'test',
        destination: 'Nairobi',
        otherDestination: '',
        departureDate: '09/27/1988',
        arrivalDate: '09/27/1988'
      }
    });
    wrapper.setProps({
      errors: ['error while creating a new request']
    });

    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
    wrapper.instance().forceUpdate();
    wrapper.find('form').simulate('submit');
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should clear the form when the component unmounts', () => {
    const componentWillUnmount = jest.spyOn(wrapper.instance(), 'componentWillUnmount');
    wrapper.unmount();
    expect(componentWillUnmount).toHaveBeenCalledTimes(1);
  });

  it('should call handleClearForm', () => {
    const wrapper = shallow(<NewRequestForm {...props} />);
    wrapper.instance().handleClearForm();
    expect(wrapper.state()).toMatchObject(defaultState);
  });


  it('should render a loading indicator while creating a new request', () => {
    wrapper.setProps({ creatingRequest: true });
    expect(wrapper.find('h5').text()).toEqual('Creating request...');
  });


  it('should save submit personal details ', () => {
    const shallowWrapper = shallow(<NewRequestForm {...props} />);
    shallowWrapper.setState({
      value: {
        name: 'tomato', // FIX: need to be refactor later
        gender: 'male',
        department: 'fame',
        role: 'job',
        manager: 'mango',
      },
      trips: [
        {}
      ]
    });
    sinon.spy(shallowWrapper.instance(), 'handleSubmit');
    shallowWrapper.instance().handleSubmit(event);
    expect(shallowWrapper.instance().handleSubmit.calledOnce).toEqual(true);
  });

  it('should have the checkbox rendered', ()=>{
    const NewRequestFormWrapper = mount(<NewRequestForm {...props} />);
    const personalDetails = NewRequestFormWrapper.find('PersonalDetailsFieldset');
    const checkBox = personalDetails.find('Checkbox');
    expect(checkBox).toHaveLength(1);
  });

  it('should add profile details to the database', () => {
    wrapper.setState({
      values: {
        name: 'tomato', // FIX: need to be refactor later
        gender: 'male',
        department: 'fame',
        role: 'job',
        manager: 'mango',
      },
      trips: [
        {}
      ]
    });
    localStorage.setItem('state', 'clicked');
    wrapper.instance().handleSubmit(event);
    expect(wrapper.prop('updateUserProfile')).toHaveBeenCalledTimes(1);
  });

});
