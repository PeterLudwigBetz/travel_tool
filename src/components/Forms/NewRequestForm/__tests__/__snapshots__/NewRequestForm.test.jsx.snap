// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewRequestForm /> renders correctly 1`] = `
<form
  className="new-request"
  onSubmit={[Function]}
>
  <fieldset
    className="personal-details"
  >
    <legend>
      Personal Details
    </legend>
    <div
      className="input-group"
    >
      <div
        className="form-input "
      >
        <label
          htmlFor="name"
        >
          Your name
          <span>
             
            (As it appears on your passport)
          </span>
        </label>
        <input
          className=""
          error=""
          label="Your name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <span
          className="error"
        />
      </div>
      <div
        className="form-input "
      >
        <label
          htmlFor="gender"
        >
          Gender
        </label>
        <div
          className="input "
        >
          <button
            className="bg-btn bg-btn--inactive"
            data-value="Female"
            key="Female"
            name="gender"
            onClick={[Function]}
            type="button"
          >
            Female
          </button>
          <button
            className="bg-btn bg-btn--inactive"
            data-value="Male"
            key="Male"
            name="gender"
            onClick={[Function]}
            type="button"
          >
            Male
          </button>
        </div>
        <span
          className="error"
        />
      </div>
      <div
        className="form-input "
      >
        <label
          htmlFor="department"
        >
          Your department
        </label>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            choices={
              Array [
                "Talent & Development",
                "Success",
                "IT",
                "Operations",
              ]
            }
            className="input select  "
            error=""
            label="Your department"
            name="department"
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            type="dropdown-select"
            value=""
          >
            <div
              className="value"
            >
              <div
                className="select-dropdown inactive"
              >
                <img
                  alt="icn"
                  src="form_select_dropdown.svg"
                />
              </div>
            </div>
            <ul
              className="select-menu select-menu--inactive"
            >
              <li
                key="Talent & Development"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Talent & Development
                </div>
              </li>
              <li
                key="Success"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Success
                </div>
              </li>
              <li
                key="IT"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  IT
                </div>
              </li>
              <li
                key="Operations"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Operations
                </div>
              </li>
            </ul>
          </div>
        </div>
        <span
          className="error"
        />
      </div>
      <div
        className="form-input "
      >
        <label
          htmlFor="role"
        >
          Your role
        </label>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            choices={
              Array [
                "Software Developer",
                "Technical Team Lead",
                "Product Engineer",
                "Product Coordinator",
                "Travel Coordinator",
                "Country Director",
              ]
            }
            className="input select  "
            error=""
            label="Your role"
            name="role"
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            type="dropdown-select"
            value=""
          >
            <div
              className="value"
            >
              <div
                className="select-dropdown inactive"
              >
                <img
                  alt="icn"
                  src="form_select_dropdown.svg"
                />
              </div>
            </div>
            <ul
              className="select-menu select-menu--inactive"
            >
              <li
                key="Software Developer"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Software Developer
                </div>
              </li>
              <li
                key="Technical Team Lead"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Technical Team Lead
                </div>
              </li>
              <li
                key="Product Engineer"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Product Engineer
                </div>
              </li>
              <li
                key="Product Coordinator"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Product Coordinator
                </div>
              </li>
              <li
                key="Travel Coordinator"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Travel Coordinator
                </div>
              </li>
              <li
                key="Country Director"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Country Director
                </div>
              </li>
            </ul>
          </div>
        </div>
        <span
          className="error"
        />
      </div>
      <div
        className="form-input "
      >
        <label
          htmlFor="manager"
        >
          Your Direct Report
          <span>
             
            (Manager)
          </span>
        </label>
        <input
          className=""
          error=""
          label="Your Direct Report"
          name="manager"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <span
          className="error"
        />
      </div>
    </div>
  </fieldset>
  <div
    className="container"
  >
    <div
      className="checkbox default"
      id="checkBox"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <img
        alt="clicked"
        className="ticksvg"
        src="tick.svg"
      />
    </div>
    <p>
      Remember these details
    </p>
  </div>
  <fieldset
    className="travel-details"
  >
    <legend>
      Travel Details
    </legend>
    <div
      className="input-group"
    >
      <div
        className="form-input "
      >
        <label
          htmlFor="origin"
        >
          Which center are you based out of?
        </label>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            choices={
              Array [
                "Kampala",
                "Lagos",
                "Nairobi",
                "New York",
              ]
            }
            className="input select  "
            error=""
            label="Which center are you based out of?"
            name="origin"
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            type="dropdown-select"
            value=""
          >
            <div
              className="value"
            >
              <div
                className="select-dropdown inactive"
              >
                <img
                  alt="icn"
                  src="form_select_dropdown.svg"
                />
              </div>
            </div>
            <ul
              className="select-menu select-menu--inactive"
            >
              <li
                key="Kampala"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Kampala
                </div>
              </li>
              <li
                key="Lagos"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Lagos
                </div>
              </li>
              <li
                key="Nairobi"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Nairobi
                </div>
              </li>
              <li
                key="New York"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  New York
                </div>
              </li>
            </ul>
          </div>
        </div>
        <span
          className="error"
        />
      </div>
      <div
        className="form-input "
      >
        <label
          htmlFor="destination"
        >
          Which center do you intend to travel to?
        </label>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            choices={
              Array [
                "Kampala",
                "Lagos",
                "Nairobi",
                "New York",
                "Other",
              ]
            }
            className="input select  "
            error=""
            label="Which center do you intend to travel to?"
            name="destination"
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            type="dropdown-select"
            value=""
          >
            <div
              className="value"
            >
              <div
                className="select-dropdown inactive"
              >
                <img
                  alt="icn"
                  src="form_select_dropdown.svg"
                />
              </div>
            </div>
            <ul
              className="select-menu select-menu--inactive"
            >
              <li
                key="Kampala"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Kampala
                </div>
              </li>
              <li
                key="Lagos"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Lagos
                </div>
              </li>
              <li
                key="Nairobi"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Nairobi
                </div>
              </li>
              <li
                key="New York"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  New York
                </div>
              </li>
              <li
                key="Other"
              >
                <div
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  Other
                </div>
              </li>
            </ul>
          </div>
        </div>
        <span
          className="error"
        />
      </div>
      <div
        className="form-input full-width other-dest--hidden"
      >
        <label
          htmlFor="otherDestination"
        >
          Specify travel destination
        </label>
        <input
          className=""
          error=""
          label="Specify travel destination"
          name="otherDestination"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <span
          className="error"
        />
      </div>
      <div
        className="form-input "
      >
        <label
          htmlFor="departureDate"
        >
          Target departure date
        </label>
        <div
          className="date-wrapper "
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className=""
                  disabled={false}
                  name="departureDate"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder=""
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <img
            alt="cal"
            className="calendar-icon"
            src="calendar_icon.svg"
          />
        </div>
        <span
          className="error"
        />
      </div>
      <div
        className="form-input "
      >
        <label
          htmlFor="arrivalDate"
        >
          Target return date
        </label>
        <div
          className="date-wrapper "
        >
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className=""
                  disabled={false}
                  name="arrivalDate"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder=""
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <img
            alt="cal"
            className="calendar-icon"
            src="calendar_icon.svg"
          />
        </div>
        <span
          className="error"
        />
      </div>
    </div>
  </fieldset>
  <hr />
  <fieldset>
    <div
      className="submit-area"
    >
      <button
        className="bg-btn bg-btn--inactive"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="bg-btn bg-btn--active"
        disabled={true}
        type="submit"
      >
        Send Request
      </button>
    </div>
  </fieldset>
</form>
`;
